internal/tls/tls.go

```diff
@@ -125,6 +129,10 @@ func NewClientConfig(opts ...Option) (*Config, error) {
 		}
 	}
 
+	if c.cfg == nil {
```

Comment:
this code is good.
but I think the following code is more simple than 132 lines.

```
c := &credentials {
    cfg: defaultTLSConfig
}

and

func WithTLSConfig(cfg *tls.Config) Option {
	return func(c *credentials) error {
		if cfg != nil {
			c.cfg = cfg
		}
		return nil
	}
}

```

Reply:
fixed :) thank you for your comment